test.disabled/ejabberd_tests/tests/rest_client_SUITE.erl

```diff
@@ -802,6 +791,106 @@ add_and_remove(Config) ->
     ok.
 
 
+add_and_remove_some_contacts_properly(Config) ->
+    escalus:fresh_story(
+        Config, [{alice, 1}, {bob, 1}, {kate, 1}, {carol, 1}],
+        fun(Alice, Bob, Kate, Carol) ->
+            BCred = credentials({bob, Bob}),
+            % adds all the other users
+            lists:foreach(fun(AddContact) -> 
+                                  add_contact_check_roster_push(AddContact, {bob, Bob}) end,
+                         [Alice, Kate, Carol]),
+            % Check Bob's roster now
+            {?OK, R2} = gett("/contacts", BCred),
```

Comment:
The correctness of contacts adding is verified in other tests. Of course it won't hurt to double-check it but in such case I think the verification part should be extracted into helper function, so it won't cloud the purpose of this test.